src/utils.jl

```diff
@@ -267,15 +268,18 @@ function _progresscallback(
     return f(dfComplete, unsafe_string(pszMessage))::Bool
 end
 
-macro _progresscallback()
-    quote
-        @cfunction(
-            $(Expr(:$, esc(_progresscallback))),
-            Cint,
-            (Cdouble, Cstring, Ptr{Cvoid})
-        )
-    end
-end
+const _cprogresscallback =
+    @cfunction(_progresscallback, Cint, (Cdouble, Cstring, Ptr{Cvoid}))
```

Comment:
Can we add a comment why this constant is being introduced? Is it due to some nuance of https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#Closure-cfunctions (to prevent something from being finalized)?